### PR TITLE
Update build.gradle setup

### PR DESCRIPTION
**Pull request by friederbluemle**
# Summary

A slightly different approach to #207 

This wraps the Android Gradle plugin dependency in the `buildscripts` section of `android/build.gradle` in a conditional:

```
if (project == rootProject) {
    // ... (dependency here)
}
```

The Android Gradle plugin is _only_ required when opening the project _stand-alone_, not when it is _included as a dependency_. By doing this, the project opens correctly in Android Studio, and it can also be consumed as a native module dependency from an application project **without** affecting the app project (avoiding unnecessary downloads/conflicts/etc).

# Test Plan

## What's required for testing (prerequisites)?

Android Studio and/or Gradle installed locally.

## What are the steps to reproduce (after prerequisites)?

Open `android` in Android Studio:

```sh
studio android/
```

Or:

Generate a Gradle wrapper on the fly, and build on command line:

```sh
cd android/
gradle wrapper --gradle-version 5.4.1 --distribution-type all
./gradlew build
```

Successfully verified and tested on:
- Android Device (Xiaomi Mi A2)
- Android Emulator (Pixel 3a)
- iOS Device (iPhone 8)
- iOS Simulator (iPhone X)

<sub><sup>Once approved, please use a *normal* GitHub merge (i.e. **NO rebase/squash** merge) to integrate the commit(s) from the PR head branch. The changes are broken up into meaningful, [atomic commits](https://www.freshconsulting.com/atomic-commits/), and my branch should already be up-to-date with the latest base branch. If it isn't, or if you want me to change anything, please let me know, and I will update the branch as soon as possible. Thank you!</sup></sub>

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -1,12 +1,19 @@
+def safeExtGet(prop, fallback) {
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
 
 buildscript {
```
Comment (dulmandakh) [inline android/build.gradle]: can we move this to if check?

Comment (friederbluemle) [reply android/build.gradle]: Sorry what do you mean?

Comment (friederbluemle) [reply android/build.gradle]: This is the same exact check that's used in **dozens** of projects (I just copy+pasted it) :)

Comment (dulmandakh) [reply android/build.gradle]: I would like to move repositories block inside the if check

Comment (friederbluemle) [reply android/build.gradle]: Ooh, I understand now. Okay sure, let me try.

Comment (friederbluemle) [reply android/build.gradle]: @dulmandakh - I updated it. I agree, it looks a bit cleaner. Thanks!

File: android/build.gradle [hunk 2]
```diff
@@ -37,22 +44,34 @@ def useDedicatedExecutor = rootProject.hasProperty('AsyncStorage_dedicatedExecut
...
 }
 
 repositories {
-    mavenCentral()
+    mavenLocal()
+    maven {
+        // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
+        url "$rootDir/../node_modules/react-native/android"
+    }
+    maven {
```
Comment (Krizzu) [inline android/build.gradle]: We don't need this

File: android/build.gradle [hunk 3]
```diff
@@ -37,22 +44,34 @@ def useDedicatedExecutor = rootProject.hasProperty('AsyncStorage_dedicatedExecut
...
-    compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
-    buildToolsVersion getExtOrDefault('buildToolsVersion')
-
+    compileSdkVersion safeExtGet('compileSdkVersion', 28)
+    buildToolsVersion safeExtGet('buildToolsVersion', '28.0.3')
     defaultConfig {
-        minSdkVersion getExtOrIntegerDefault('minSdkVersion')
-        targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
-
+        minSdkVersion safeExtGet('minSdkVersion', 19)
```
Comment (Krizzu) [inline android/build.gradle]: I'm happy keeping 19 here, but as @dulmandakh said, it would be better to comply with RN's 16.

Comment (friederbluemle) [reply android/build.gradle]: Sure, I agree it would be best to comply with 16 from react-native. Is it okay to make this change in a separate PR? Lowering the minimum SDK _could_ require other code changes. Since this PR is not about change the SDK level, it would keep these changes more focussed (and easier to review).

File: android/build.gradle (hunk 2 above)
Comment (friederbluemle) [reply android/build.gradle]: Sure. I removed it.

**Comment by friederbluemle** [comment]
@Krizzu - Thanks for the review. I addressed your comments, and also removed the two "getExtOrDefault" (I thought I removed them earlier, but missed it ;))
Regarding opening it stand-alone: Sometimes it can be useful to validate and compile the project in isolation, without any other projects or apps involved. It would also be possible to add unit tests that specifically target library code (and of course don't need an app project).

File: android/build.gradle [hunk 4]
```diff
@@ -1,23 +1,22 @@
...
+    // The Android Gradle plugin is only required when opening the android folder stand-alone.
+    // This avoids unnecessary downloads and potential conflicts when the library is included as a
+    // module dependency in an application project.
+    if (project == rootProject) {
+        repositories {
+            google()
+            jcenter()
+        }
+        dependencies {
+            classpath 'com.android.tools.build:gradle:3.4.1'
```
Comment (friederbluemle) [inline android/build.gradle]: @Krizzu - Quick question: I kept the version here the same as before, but could also update it to the latest stable version (`3.5.0`) at the same time. It would have no impact on any consumer app project. OK to make the change? Or do you prefer it to be in a separate PR?

Comment (Krizzu) [reply android/build.gradle]: Yes, please upgrade to 3.5.0 in this PR

File: android/build.gradle (hunk 3 above)
Comment (Krizzu) [reply android/build.gradle]: You're right, let's keep 19 for now

File: android/build.gradle (hunk 4 above)
Comment (friederbluemle) [reply android/build.gradle]: Done.

**Comment by Krizzu** [comment]
Thanks a lot @friederbluemle 	@dulmandakh
